hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/utils/AutoCompletion.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.utils;
+
+import java.util.Objects;
+import org.apache.hadoop.hdds.cli.GenericCli;
+import org.apache.ratis.util.ReflectionUtils;
+import org.reflections.Reflections;
+import picocli.AutoComplete;
+import picocli.CommandLine;
+import picocli.CommandLine.Option;
+
+/**
+ * Tool to generate bash/zsh auto-complete scripts for Ozone CLI.
+ */
+public final class AutoCompletion {
```

Comment:
I think we should make this a `@Command` with the usual options and extend `GenericCli`.

Benefits:

- `ozone completion --help` would work as expected, now it just ignores the option and generates the completion script to the console
- `ozone completion` itself would have auto-completion
- we could add an option for writing output to a file (`--output-file <filename>`)

Reply:
Thanks for the suggestion, will update the PR.

Reply:
Please feel free to ignore the point about `--output-file`.